website/test.ts

```diff
@@ -0,0 +1,208 @@
+  assertEqual(D04.dataType.type, "IndexedAccessTypeNode");
+  assertEqual(D04.dataType.index.type, "TypeReference");
+  assertEqual(D04.dataType.index.name, "key");
+  assertEqual(D04.dataType.index.arguments, undefined);
+  assertEqual(D04.dataType.object.type, "TypeReference");
+  assertEqual(D04.dataType.object.name, "Y");
+  assertEqual(D04.dataType.object.arguments, undefined);
+
+  // Test T05
+  const T05 = docs[5];
+  assertEqual(T05.name, "T05");
+  assertEqual(T05.documentation, "comment 8");
+  // Test definitions
+  const D05 = T05.definition;
+  assertEqual(D05.type, "TypeOperator");
+  assertEqual(D05.operator, "KeyOfKeyword");
+  assertEqual(D05.subject.type, "ParenthesizedType");
+  assertEqual(D05.subject.elementType.type, "IntersectionType");
+  assertEqual(D05.subject.elementType.types.length, 2);
+  // Test definition element type types
+  assertEqual(D05.subject.elementType.types[0].type, "TypeReference");
+  assertEqual(D05.subject.elementType.types[0].name, "A");
+  assertEqual(D05.subject.elementType.types[0].arguments, undefined);
+  assertEqual(D05.subject.elementType.types[1].type, "TypeReference");
+  assertEqual(D05.subject.elementType.types[1].name, "B");
+  assertEqual(D05.subject.elementType.types[1].arguments, undefined);
+  // Skipping parameters, tested several times.
+
+  // Test T06
+  const T06 = docs[6];
+  assertEqual(T06.name, "T06");
+  assertEqual(T06.documentation, "comment 9");
+  // Test definition.
+  const D06 = T06.definition;
+  assertEqual(D06.type, "ConditionalType");
+  // Test checkType
+  assertEqual(D06.checkType.type, "TypeReference");
+  assertEqual(D06.checkType.name, "X");
+  // Test extendsType
+  assertEqual(D06.extendsType.type, "TypeReference");
+  assertEqual(D06.extendsType.name, "Y");
+  // Test falseType
+  assertEqual(D06.falseType.type, "keyword");
+  assertEqual(D06.falseType.name, "string");
+  // Test trueType
+  assertEqual(D06.trueType.type, "keyword");
+  assertEqual(D06.trueType.name, "number");
+  // Skipping parameters, tested several times.
+
+  // Test T07
+  const T07 = docs[7];
+  assertEqual(T07.name, "T07");
+  assertEqual(T07.documentation, "comment 10");
+  // Test definition
+  const D07 = T07.definition;
+  assertEqual(D07.type, "MappedType");
+  // No need to test D07.parameter completely, just check if it contains that.
+  assertEqual(D07.parameter.type, "TypeParameter");
+  assertEqual(D07.parameter.name, "P");
+  // IndexedAccessTypeNode has been already tested
+  assertEqual(D07.dataType.type, "IndexedAccessTypeNode");
+
+  // Test T08
+  const T08 = docs[8];
+  assertEqual(T08.name, "T08");
+  assertEqual(T08.documentation, "comment 11");
+  assertEqual(T08.definition.dataType.type, "UnionType");
+  assertEqual(T08.definition.dataType.types.length, 2);
+  assertEqual(T08.definition.dataType.types[1].type, "keyword");
+  assertEqual(T08.definition.dataType.types[1].name, "null");
+});
+
+test(async function test_removeSpaces() {
```

Comment:
Move to `util_test.ts` and maybe rename this file to `gendoc_test.ts` ?

Reply:
I prefer to rename this file and move `util.ts` to `gendoc_util.ts`.
(it's just one part of the website, and I think having two test files for it a lot)

Reply:
I appreciate the strive for minimalism, but this has already expanded into multiple files - so I think it's better to have unit tests for each file.ts in a corresponding file_test.ts. But this is all nitpicky - the most important step now is to get `testdoc/something.js` working.

Reply:
done